function/pkg/operator/operator_test.go

```diff
@@ -483,6 +519,368 @@ func Test_waitForObject(t *testing.T) {
+
+	configurationObjectWithChangedElementOfMetadataAnnotations := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithChangedElementOfMetadataAnnotations.Object["metadata"].(map[string]interface{})["annotations"].(map[string]interface{})["xx"] = "abcd"
+
+	configurationObjectWithChangedElementOfSpec := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithChangedElementOfSpec.Object["spec"].(map[string]interface{})["test"] = "abcd"
+
+	configurationObjectWithAddedElementOfMetadataLabels := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithAddedElementOfMetadataLabels.Object["metadata"].(map[string]interface{})["labels"].(map[string]interface{})["new"] = "abcd"
+
+	configurationObjectWithAddedElementOfMetadataAnnotations := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithAddedElementOfMetadataAnnotations.Object["metadata"].(map[string]interface{})["annotations"].(map[string]interface{})["new"] = "abcd"
+
+	configurationObjectWithAddedElementOfSpec := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithAddedElementOfSpec.Object["spec"].(map[string]interface{})["new"] = "abcd"
+
+	configurationObjectWithRemovedElementOfMetadataLabels := *(defaultConfigurationObject.DeepCopy())
+	delete(configurationObjectWithRemovedElementOfMetadataLabels.Object["metadata"].(map[string]interface{})["labels"].(map[string]interface{}), "cc")
+
+	configurationObjectWithRemovedElementOfMetadataAnnotations := *(defaultConfigurationObject.DeepCopy())
+	delete(configurationObjectWithRemovedElementOfMetadataAnnotations.Object["metadata"].(map[string]interface{})["annotations"].(map[string]interface{}), "vv")
+
+	configurationObjectWithRemovedElementOfSpec := *(defaultConfigurationObject.DeepCopy())
+	delete(configurationObjectWithRemovedElementOfSpec.Object["spec"].(map[string]interface{})["subscriber"].(map[string]interface{})["ref"].(map[string]interface{}), "kind")
+
+	configurationObjectWithNilMetadataLabels := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithNilMetadataLabels.Object["metadata"].(map[string]interface{})["labels"] = nil
+
+	configurationObjectWithNilMetadataAnnotations := *(defaultConfigurationObject.DeepCopy())
+	configurationObjectWithNilMetadataAnnotations.Object["metadata"].(map[string]interface{})["annotations"] = nil
+
+	type args struct {
+		first  unstructured.Unstructured
+		second unstructured.Unstructured
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "the same configuration objects",
+			args: args{
+				first:  defaultConfigurationObject,
+				second: defaultConfigurationObject,
+			},
+			want: true,
+		},
+		{
+			name: "similar configuration objects - changed metadata/^(labels|annotations)/*",
+			args: args{
+				first:  defaultConfigurationObject,
+				second: similarConfigurationObjectWithChangedOmmitedPartOfMetadata,
```

Comment:
I see here possible two small improvements. 
1. Name of this field is definitely too long :D To be honest if the longest name I've ever seen
2. Instead of doing this operation above test tables you can use the `func() any_tyle { ... }()` anonymous method to keep this logic close to a specific test case

Reply:
regarding this first point, I can share with you two good materials about code style and good practices in go:
1. https://go.dev/doc/effective_go <- official one
2. https://github.com/uber-go/guide/blob/master/style.md <- I really like this one

Reply:
ad 1 (from first comment): I have no idea how to change these names without losing their meaning. I prefer long names than `testObj` and` testObj1`.

ad 2: All configuration objects prepared for tests that are not reused have been moved to built-in functions.

Reply:
When I have variable `invoice` I know that is something connected with invoice. When I see `testObj` I know only that it's something for tests. 

IMO when I have `destinationConfigurationObjectWithoutLabelsAndAnnotations` I know that it is specific object described in name and I see it in every place where I use it (not only in place with comment). Now we have no problem with too small screens and may use long names. I prefer it but I'm curious arguments against it.

Reply:
We talked about this and decided that I will remove these long named objects because reusing them creates a coupling between different tests.

Reply:
Done